types/firestore.d.ts

```diff
@@ -9242,6 +9354,59 @@ declare namespace FirebaseFirestore {
     sort(...orderings: Ordering[]): Pipeline<AppModelType>;
     sort(options: {orderings: Ordering[]}): Pipeline<AppModelType>;
 
+    /**
+     * Performs union of all documents from two pipelines, including duplicates.
+     *
+     * <p>This stage will pass through documents from previous stage, and also pass through documents
+     * from previous stage of the `other` {@code Pipeline} given in parameter. The order of documents
+     * emitted from this stage is undefined.
+     *
+     * <p>Example:
+     *
+     * <pre>{@code
+     * // Emit documents from books collection and magazines collection.
+     * firestore.pipeline().collection("books")
+     *     .union(firestore.pipeline().collection("magazines"));
+     * }</pre>
+     *
+     * @param other The other {@code Pipeline} that is part of union.
+     * @return A new {@code Pipeline} object with this stage appended to the stage list.
+     */
+    union(other: Pipeline<AppModelType>): Pipeline<AppModelType>;
+
+    /**
+     * Produces a document for each element in array found in previous stage document.
+     *
+     * <p>For each previous stage document, this stage will emit zero or more augmented documents. The
+     * input array found in the previous stage document field specified by the `fieldName` parameter,
+     * will for each input array element produce an augmented document. The input array element will
+     * augment the previous stage document by replacing the field specified by `fieldName` parameter
+     * with the element value.
+     *
+     * <p>In other words, the field containing the input array will be removed from the augmented
+     * document and replaced by the corresponding array element.
+     *
+     * <p>Example:
+     *
+     * <pre>{@code
+     * // Input:
+     * // { "title": "The Hitchhiker's Guide to the Galaxy", "tags": [ "comedy", "space", "adventure" ], ... }
+     *
+     * // Emit a book document for each tag of the book.
+     * firestore.pipeline().collection("books")
+     *     .unnest("tags");
+     *
+     * // Output:
+     * // { "title": "The Hitchhiker's Guide to the Galaxy", "tags": "comedy", ... }
+     * // { "title": "The Hitchhiker's Guide to the Galaxy", "tags": "space", ... }
+     * // { "title": "The Hitchhiker's Guide to the Galaxy", "tags": "adventure", ... }
+     * }</pre>
+     *
+     * @param field The name of the field containing the array.
+     * @return A new {@code Pipeline} object with this stage appended to the stage list.
+     */
+    unnest(field: Selectable | string): Pipeline<AppModelType>;
```

Comment:
Do we also need a declaration for the `unnest(options: UnnestOptions)` overload?